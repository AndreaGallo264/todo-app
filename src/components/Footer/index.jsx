import React from "react";
import { AppBar, makeStyles, Fab } from "@material-ui/core";
import AddIcon from "@material-ui/icons/Add";
import { Link } from "react-router-dom";
import { DarkTooltip } from "../DarkTooltip";
const useStyles = makeStyles((theme) => ({
  appBar: {
    top: "auto",
    bottom: 0,
  },
  fabButton: {
    position: "absolute",
    zIndex: 1,
    top: -70,
    left: 0,
    right: 0,
    margin: "0 auto",
  },
}));
const Footer = () => {
  const classes = useStyles();
  return (
    <AppBar position="fixed" color="primary" className={classes.appBar}>
      <Link to="/new">
        <DarkTooltip title="Add">
          <Fab color="secondary" className={classes.fabButton}>
            <AddIcon />
          </Fab>
        </DarkTooltip>
      </Link>
    </AppBar>
  );
};

export default Footer;
