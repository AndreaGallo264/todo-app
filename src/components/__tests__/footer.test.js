import React from "react";
import renderer from "react-test-renderer";
import Footer from "../Footer";
import { BrowserRouter as Router } from "react-router-dom";

describe("Footer component", () => {
  test("Matches snapshot", () => {
    const tree = renderer
      .create(
        <Router>
          <Footer />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
