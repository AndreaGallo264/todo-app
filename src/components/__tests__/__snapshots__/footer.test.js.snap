// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Matches snapshot 1`] = `
<footer>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-1 mui-fixed MuiPaper-elevation4"
  >
    <a
      href="/new"
      onClick={[Function]}
    >
      <button
        aria-describedby={null}
        className="MuiButtonBase-root MuiFab-root makeStyles-fabButton-2 MuiFab-secondary"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Add"
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </a>
  </header>
</footer>
`;
